languages/cpp/examples/Wrapper.cpp

```diff
@@ -24,51 +28,91 @@ int main() {
     bitwardenSettings.set_identity_url(identityUrl);
 
     // Create a Bitwarden client instance
-    BitwardenClient bitwardenClient = BitwardenClient(bitwardenSettings);
-    // // Access token login
-    bitwardenClient.loginAccessToken(accessToken);
-    // Organization ID
-    boost::uuids::uuid organizationUuid = boost::uuids::string_generator()(organizationId);    
-    
-    // // Create a new project
+    BitwardenClient bitwardenClient(bitwardenSettings);
+
+    // Access token login
+    bitwardenClient.loginAccessToken(accessToken, stateFile);
+
+    // Convert organization ID to UUID
+    boost::uuids::uuid organizationUuid = boost::uuids::string_generator()(organizationId);
+
+    // Create a new project
+    std::cout << "Projects:\n";
     ProjectResponse projectResponseCreate = bitwardenClient.createProject(organizationUuid, "NewTestProject");
     boost::uuids::uuid projectId = boost::uuids::string_generator()(projectResponseCreate.get_id());
-   
+
+    std::cout << "\tcreateProject: '" << projectResponseCreate.get_name() << "'\n\n";
+
     // List projects
     ProjectsResponse projectResponseList = bitwardenClient.listProjects(organizationUuid);
+    std::cout << "\tlistProjects:\n";
+    for (const ProjectResponse& project : projectResponseList.get_data()) {
+        std::cout << "\t\tID: '" << project.get_id() << "', Name: '" << project.get_name() << "'\n";
+    }
+    std::cout << '\n';
 
     // Get project details
     ProjectResponse projectResponseGet = bitwardenClient.getProject(projectId);
+    std::cout << "\tgetProject:\n\t\tID: '" << projectResponseGet.get_id() << "', Name: '" << projectResponseGet.get_name() << "'\n\n";
 
     // Update project
-    ProjectResponse ProjectResponseUpdate = bitwardenClient.updateProject(projectId, organizationUuid, "NewTestProject2");
+    ProjectResponse projectResponseUpdate = bitwardenClient.updateProject(organizationUuid, projectId, "NewTestProject2");
+    std::cout << "\tupdateProject: '" << projectResponseUpdate.get_name() << "'\n\n";
 
     // Secrets
     std::string key = "key";
     std::string value = "value";
     std::string note = "note";
 
+    // Sync secrets
+    std::cout << "Secrets:\n";
+    std::cout << "\tSyncing secrets...\n";
+    std::chrono::system_clock::time_point lastSyncedDate = std::chrono::system_clock::now();
+    SecretsSyncResponse secretsSyncResponse = bitwardenClient.sync(organizationUuid, lastSyncedDate);
```

Comment:
How about we set the lastSyncedDate to be optional here ?
And move the `std::chrono::system_clock::now();` to be at the second sync `// Sync changes to secrets`

Reply:
I updated the sync example to do an initial sync with an empty lastSyncedDate and a second, redundant sync to demonstrate that the sync "has changes" behavior works correctly in 3a7cd09e9d1d7fcf19ef5c9664552cd552ad9dcb.

Reply:
I left the `std::chrono::system_clock::now();` to occur after the first sync though, as I think it more accurately reflects when the last sync occurred. Please let me know if that doesn't make sense.